src/OrthoViewController.cc

```diff
@@ -0,0 +1,289 @@
+
+  auto screenPos = this->dataPtr->camera->Project(
+    this->dataPtr->target);;
+
+  translation.Set(0.0,
+      ((width/2.0 - screenPos.X()) / static_cast<float>(width))
+      * orthoWidth,
+      ((height/2.0 - screenPos.Y()) / static_cast<float>(height))
+      * orthoHeight);
+
+  this->dataPtr->camera->SetWorldPosition(
+      this->dataPtr->camera->WorldPosition() +
+      this->dataPtr->camera->WorldRotation() * translation);
+
+  // zoom by changing scale
+  // for simplicity, zoom in if a positive value is given otherwise
+  // zoom out
+  // todo(anyone) improving zooming based on distance to target
+  double factor = 1.0;
+  if (_value > 0)
+    factor = 0.9;
+  else
+    factor = 1.1;
+
+  this->dataPtr->scale /= factor;
+
+  // build custom projection matrix from custom near and far clipping planes,
+  // had to set a negative near clippping plane to workaround a camera
+  // culling issue in orthographic view.
+  auto proj = this->dataPtr->BuildScaledOrthoMatrix(
+      -width / this->dataPtr->scale / 2.0,
+       width / this->dataPtr->scale / 2.0,
+      -height / this->dataPtr->scale / 2.0,
+       height / this->dataPtr->scale / 2.0,
+      -500, this->dataPtr->camera->FarClipPlane());
+
+  this->dataPtr->camera->SetProjectionMatrix(proj);
+
+  double newOrthoWidth = width / this->dataPtr->scale;
+  double newOrthoHeight = height / this->dataPtr->scale;
+
+  translation.Set(0.0,
+      ((screenPos.X() - width/2.0) / static_cast<double>(width))
+      * newOrthoWidth,
+      ((screenPos.Y() - height/2.0) / static_cast<double>(height))
+      * newOrthoHeight);
+
+//  this->TranslateLocal(translation);
+  this->dataPtr->camera->SetWorldPosition(
+      this->dataPtr->camera->WorldPosition() +
+      this->dataPtr->camera->WorldRotation() * translation);
+}
+
+//////////////////////////////////////////////////
+void OrthoViewController::Resize(unsigned int _width, unsigned int _height)
+{
+  math::Matrix4d proj = this->dataPtr->BuildScaledOrthoMatrix(
+       _width / this->dataPtr->scale / -2.0,
+       _width / this->dataPtr->scale / 2.0,
+       _height / this->dataPtr->scale / -2.0,
+       _height / this->dataPtr->scale / 2.0,
+      -500, this->dataPtr->camera->FarClipPlane());
+
+  this->dataPtr->camera->SetProjectionMatrix(proj);
+}
+
+//////////////////////////////////////////////////
+void OrthoViewController::Pan(const math::Vector2d &_value)
+{
+  if (!this->dataPtr->camera)
+  {
+    ignerr << "Camera is NULL" << std::endl;
+    return;
+  }
+
+  double viewportWidth = this->dataPtr->camera->ImageWidth();
+  double viewportHeight = this->dataPtr->camera->ImageHeight();
+
+//  double distance =
+//      this->dataPtr->camera->WorldPosition().Distance(this->dataPtr->target);
+
+  // pseudo distance
+//   double distance = 1000.0/this->dataPtr->scale;
+
+//  double hfov = this->dataPtr->camera->HFOV().Radian();
+//  double vfov = 2.0f * atan(tan(hfov / 2.0f) /
+//        this->dataPtr->camera->AspectRatio());
```

Comment:
Reminder to remove these comments if they're not needed

Reply:
removed 6e2e25f